### PR TITLE
NIFIREG-256 Updating manage buckets page

### PR DESCRIPTION
**Pull request by bbende**

### CODE REVIEW

**Comment by andrewmlim** [comment]
Reviewed from a UI/UX perspective.

registry_bucket_config

- I think it would be really helpful to add a tooltip to the "Allow Bundle Redeploy" checkbox
- Should add some documentation around the new checkbox (can be a separate PR)

Looking at the User config in comparison and seeking consistency with it:

user_config


- Perhaps do some text formatting for the Bucket ID and the Allow Bundle Redeploy button
- Checkbox selections are saved automatically in the User Config.  But those selections have to do with the policies on the user, not the user him/herself.  Allow redeploy is an update to the bucket itself.  Will want the UI to make this distinction as clear as possible.

@moranr any thoughts on this last two bullets?

**Comment by andrewmlim** [comment]
Just wanted to note that I filed https://issues.apache.org/jira/browse/NIFIREG-257 which requests that any changes to Special Privileges in User configuration are acknowledged by the UI by the "Success"  toast message. So the same behavior should apply to the checkbox here in Bucket configuration.

**Comment by moranr** [comment]
I wanted to suggest a few changes. Based on @andrewmlim 's previous comment re: NIFIREG-257, make the bundle redeploy checkbox behave the same way. I think adding a heading above the option will make it feel like its own section and not like it's just floating off by itself. I tried to come up with a bit more meaningful text _("Allow bundle overwrite")_ and think adding a tooltip will help as well. Anyone have ideas for the tooltip message?

Also, an idea for the bucket ID information: move this kind of reference-type detail to the bottom, allowing quicker access to editable things up top. I think this pattern can be repeated in other areas of the app as well and prove useful when there is a lot more content to display.

A quick mockup for reference:
NIFIREG-256 Updating manage buckets page #174

**Comment by bbende** [comment]
Thanks @moranr and @andrewmlim , will work on the updates based on Rob's mock-up

**Comment by bbende** [comment]
@andrewmlim I pushed changes to the branch that should make it works as shown in the screenshot.

I force-pushed so you'll have to reset your branch.

**Comment by andrewmlim** [comment]
@bbende Looked at your latest changes. Really like how checking/unchecking the "Allow bundle overwrite" is automatically saved now.

For consistency with the User config sidenav, we should have the "?" icon to activate the tooltip for the checkbox.  Right now, the tooltip shows up when you hover over the text.

One nitpick: can you tighten up the spacing before and after "Bundle Settings"?  Same with the spacing after "About".  See the mockup by @moranr for reference.

**Comment by bbende** [comment]
@andrewmlim pushed another commit that addresses your suggestions

**Comment by andrewmlim** [comment]
@bbende Latest changes look good!

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.html [hunk 1]
```diff
@@ -41,6 +41,20 @@
...
             </button>
         </div>
+        <div class="pad-top-md pad-bottom-sm pad-left-md pad-right-md">
+            <div class="pad-bottom-sm">
+                <span class="md-card-title">Bundle Settings</span>
+            </div>
+            <div>
+                <mat-checkbox #bundleRedeployCheckbox
+                              [disabled]="!nfRegistryService.currentUser.resourcePermissions.buckets.canWrite"
+                              [(checked)]="nfRegistryService.bucket.allowBundleRedeploy"
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.html]: Please remove this line.

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.html [hunk 2]
```diff
@@ -41,6 +41,20 @@
...
+                              [(ngModel)]="allowBundleRedeploy"
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.html]: I found a slight bug in this PR. If you edit a bucket and check to "Allow bundle overwrite" for said bucket... and then you click the 'X' in the top right of the sidenav... and then you click the browser's back navigation button... you will see that the checkbox selection is incorrect (it should be checked at this point). To fix this please remove the ```[(checked)]="nfRegistryService.bucket.allowBundleRedeploy"``` and update the ```[(ngModel)]="nfRegistryService.bucket.allowBundleRedeploy"```

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js [hunk 3]
```diff
@@ -413,15 +417,49 @@ NfRegistryManageBucket.prototype = {
...
+                    title: 'Success',
+                    message: 'Bundle settings have been updated.',
+                    verticalPosition: 'bottom',
+                    horizontalPosition: 'right',
+                    icon: 'fa fa-check-circle-o',
+                    color: '#1EB475',
+                    duration: 3000
+                });
+            }  else if (response.status === 400) {
+                self.allowBundleRedeploy = self.nfRegistryService.bucket.allowBundleRedeploy;
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js]: self.allowBundleRedeploy = !event.checked

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js [hunk 4]
```diff
@@ -101,7 +102,7 @@ NfRegistryManageBucket.prototype = {
                     self.nfRegistryService.sidenav.open();
                     var bucket = response[0];
                     self.nfRegistryService.bucket = bucket;
-                    self._bucketname = response[0].name;
+                    self.bucketname = response[0].name;
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js]: self.allowBundleRedeploy = bucket.allowBundleRedeploy;

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js [hunk 5]
```diff
@@ -413,15 +417,49 @@ NfRegistryManageBucket.prototype = {
...
+                self.bucketname = self.nfRegistryService.bucket.name;
                 self.dialogService.openConfirm({
                     title: 'Error',
                     message: 'This bucket already exists. Please enter a different identity/bucket name.',
                     acceptButton: 'Ok',
                     acceptButtonColor: 'fds-warn'
                 });
             } else if (response.status === 400) {
-                self._bucketname = self.nfRegistryService.bucket.name;
+                self.bucketname = self.nfRegistryService.bucket.name;
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js]: self.allowBundleRedeploy = self.nfRegistryService.bucket.allowBundleRedeploy;

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js [hunk 6]
```diff
@@ -413,15 +417,49 @@ NfRegistryManageBucket.prototype = {
                     duration: 3000
                 });
             } else if (response.status === 409) {
-                self._bucketname = self.nfRegistryService.bucket.name;
+                self.bucketname = self.nfRegistryService.bucket.name;
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js]: self.allowBundleRedeploy = self.nfRegistryService.bucket.allowBundleRedeploy;

**Comment by bbende** [comment]
@scottyaslan thanks for the review! I pushed another commit that should incorporate the changes you suggested.

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js (hunk 6 above)
Comment (scottyaslan) [reply nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js]: You forgot to reset it here in the case of a 409...

**Pull request closed without merging**